syntax = "proto2";

package topology;

message BlockProto {
  optional int32 port_speed = 1 [default = 10];  // Unit is Gbps.
  optional int32 radix = 2;
}

message PhysicalLinkProto {
  optional int32 ocs_index = 1;
  optional int32 block_index = 2;
  optional int32 num_ocs_block_links = 3;
  optional int32 num_block_ocs_links = 4;
}

message LogicalLinkProto {
  optional int32 block1_index = 1;
  optional int32 block2_index = 2;
  optional int32 num_block1_block2_links = 3;
  optional int32 num_block2_block1_links = 4;
}

message PhysicalTopology {
  repeated BlockProto blocks = 1;
  optional int32 num_ocses = 2;
  repeated PhysicalLinkProto links = 3;
}

message LogicalTopology {
  repeated BlockProto blocks = 1;
  repeated LogicalLinkProto links = 2;
}

